### ISSUE TITLE
Removed shebang breaks shims in Fish

### ISSUE THREAD

**Issue by icollar** [pyenv/pyenv#1336]
Shims are no longer working for Fish due to https://github.com/pyenv/pyenv/pull/1153, this is currently an issue on the master branch, which is the default version installed by pyenv-installer.

**Works fine in Bash:**
```
icollar@1804:~$ pyenv shell 3.5.7
icollar@1804:~$ python
Python 3.5.7 (default, Apr 23 2019, 10:40:07) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 
```

**Fails in Fish:**
```
~  pyenv shell 3.5.7
~  python
Failed to execute process '/home/icollar/.pyenv/shims/python'. Reason:
exec: Exec format error
The file '/home/icollar/.pyenv/shims/python' is marked as an executable but could not be run by the operating system.
```

**Warnings in zsh:**
```
icollar@1804 ~ % pyenv shell 3.5.7     
icollar@1804 ~ % python
/home/icollar/.pyenv/shims/python: 5: /home/icollar/.pyenv/shims/python: [[: not found
Python 3.5.7 (default, Apr 23 2019, 10:40:07) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 
```

**Comment by mpyrev** [comment]
I have the same warning message as zsh in bash (Ubuntu 16.04).

**Comment by zr40** [comment]
This actually breaks when called through a process manager such as supervisord:

```
supervisor: couldn't exec /home/app/.pyenv/shims/python3: ENOEXEC
supervisor: child process was not spawned
```

**Comment by nvllsvm** [comment]
Also breaks virtualenv
```
Traceback (most recent call last):
  File "/Users/andrewr/.local/pythonuser/bin/virtualenv", line 10, in <module>
    sys.exit(main())
  File "/Users/andrewr/.local/pythonuser/lib/python3.7/site-packages/virtualenv.py", line 750, in main
    sub_process_call = subprocess.Popen([interpreter, file] + sys.argv[1:], env=env)
  File "/Users/andrewr/.local/git/pyenv/versions/3.7.3/lib/python3.7/subprocess.py", line 775, in __init__
    restore_signals, start_new_session)
  File "/Users/andrewr/.local/git/pyenv/versions/3.7.3/lib/python3.7/subprocess.py", line 1522, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
OSError: [Errno 8] Exec format error: '/Users/andrewr/.local/git/pyenv/shims/python'
```

**Comment by chrahunt** [comment]
Fixed by #1337.
